### PR TITLE
Annotations generators: Add linter coverage

### PR DESCRIPTION
**Pull request by orelmisan**
### What this PR does
PRs #12558 and #12710 introduced the following packages without linter covarage:
- pkg/network/pod/annotations
- pkg/storage/pod/annotations

Add linter coverage for both packages.

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [X] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [X] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
None
```

### CODE REVIEW

File: pkg/storage/pod/annotations/generator_test.go [hunk 1]
```diff
@@ -39,10 +39,12 @@ var _ = Describe("Annotations Generator", func() {
 		Expect(err).NotTo(HaveOccurred())
 
 		expectedAnnotations := map[string]string{
-			"pre.hook.backup.velero.io/container":  "compute",
+			"pre.hook.backup.velero.io/container": "compute",
+			//nolint: lll
```
Comment (EdDev) [inline pkg/storage/pod/annotations/generator_test.go:43]: I prefer this is solved and not ignored. If you think there is a very (very!) good reason to do so, please convince us on the commit message.

Comment (orelmisan) [reply pkg/storage/pod/annotations/generator_test.go]: Done.

**Comment by orelmisan** [comment]
[Change](https://github.com/kubevirt/kubevirt/compare/cf94adc6ed4b98dbe70bf074013c12d485dee706..8662c438274b94250177a1eccdaad1ee7187c5d5): Answered @EdDev's [comment](https://github.com/kubevirt/kubevirt/pull/12846#discussion_r1763133783).

**Review by EdDev — APPROVED** [review]
Thanks!

Approving the network part:
/approve

**Pull request merged**